libnavui-maneuver/src/main/java/com/mapbox/navigation/ui/maneuver/LaneIconProcessor.kt

```diff
@@ -119,6 +166,17 @@ internal object LaneIconProcessor {
                     }
                 }
             }
+            laneIndicator.directions.size > 1 &&
+                laneIndicator.directions.contains(SLIGHT_LEFT) &&
+                laneIndicator.directions.contains(SLIGHT_RIGHT) -> {
```

Comment:
This combination also needs a special case for straight.

Reply:
Do you mean 3 lane combinations? Like Slight Left + Straight + Slight Right.
If yes then I have already added 
```
laneIndicator.directions.size > 2 &&
                laneIndicator.directions.contains(SLIGHT_LEFT) &&
                laneIndicator.directions.contains(SLIGHT_RIGHT) &&
                laneIndicator.directions.contains(STRAIGHT)
```

Reply:
For this particular [route](https://hey.mapbox.com/directions-debug/#route=-122.42337642876043,37.77166572535229;-122.40625482901194,37.771762050234145&server=https://api.mapbox.com&profile=mapbox/driving-traffic&style=empty&queryparams=steps%3Dtrue%26overview%3Dfull%26geometries%3Dgeojson%26roundabout_exits%3Dtrue%26voice_units%3Dimperial%26voice_instructions%3Dtrue%26banner_instructions%3Dtrue&token=pk.eyJ1Ijoic3ZjLW9rdGEtbWFwYm94LXN0YWZmLWFjY2VzcyIsImEiOiJja2d0bHMxbDcwNHI0MnRxbHBwNTBoeHN0In0.X5Gn0apaqXAXDkZ1sSDLfA) if you would look at the first banner instruction, you will see a `sub` with lanes. The 1st lane is an active left, the 2nd lane is slight left and slight right with active slight left and 3rd lane is inactive slight right. However, Android SDK does not uses `sub.components[x].active_direction` to find out the valid indication. This is because `sub.components[x].active_direction` is not available for all profiles(CYCLING). Hence we use `primary.modifier`. So, now if you look `primary.modifier`, you would see the valid indication is `left`. Hence the below will go to the else logic which is incorrect.
```
laneIndicator.directions.size > 1 &&
    laneIndicator.directions.contains(SLIGHT_LEFT) &&
    laneIndicator.directions.contains(SLIGHT_RIGHT) -> {
        activeDirection?.let {
            if (it == SLIGHT_LEFT) {
                laneIcon[LaneTurns.LANE_SLIGHT_LEFT_SLIGHT_RIGHT_SLIGHT_LEFT_ONLY]
            } else {
                laneIcon[LaneTurns.LANE_SLIGHT_LEFT_SLIGHT_RIGHT_SLIGHT_RIGHT_ONLY]
            }
        }
}
```

cc @1ec5

Reply:
Hence maybe the logic should be:

```
laneIndicator.directions.size > 1 &&
    laneIndicator.directions.contains(SLIGHT_LEFT) &&
    laneIndicator.directions.contains(SLIGHT_RIGHT) -> {
        activeDirection?.let {
            if (it == SLIGHT_LEFT || LEFT) {
                laneIcon[LaneTurns.LANE_SLIGHT_LEFT_SLIGHT_RIGHT_SLIGHT_LEFT_ONLY]
            } else if (it == SLIGHT_RIGHT || RIGHT) {
                laneIcon[LaneTurns.LANE_SLIGHT_LEFT_SLIGHT_RIGHT_SLIGHT_RIGHT_ONLY]
            } else {
                laneIcon[LaneTurns.LANE_STRAIGHT]
            }
        }
}
```
Wdyt?

Reply:
Okay, I changed the logic as above

Reply:
Oh, OK, I must’ve missed your original logic that accounted for the straight indication. 👍